### PR TITLE
fix: remove sensitive information from agent config toJSON() method

### PR DESCRIPTION
**Pull request by jimezesinachi**
Signed-off-by: Jim Ezesinachi <jim@animo.id>

### CODE REVIEW

File: packages/core/src/agent/AgentConfig.ts [hunk 1]
```diff
@@ -191,8 +191,13 @@ export class AgentConfig {
...
     return {
       ...this.initConfig,
-      logger: this.logger !== undefined,
-      agentDependencies: this.agentDependencies != undefined,
+      walletConfig: {
+        ...this.walletConfig,
+        key: undefined,
+        storage: { ...this.walletConfig?.storage, credentials: undefined },
+      },
+      logger: Boolean(this.logger), // Maybe show the loglevel here instead of true / false?
```
Comment (blu3beri) [inline packages/core/src/agent/AgentConfig.ts:199]: I think we can just log the level here. `this.logger.logLevel` I think.

**Review by blu3beri — CHANGES_REQUESTED** [review]

**Comment by rapaktech** [comment]
All checks passed! @blu3beri

File: packages/core/src/agent/AgentConfig.ts [hunk 2]
```diff
@@ -191,8 +191,13 @@ export class AgentConfig {
...
       ...this.initConfig,
-      logger: this.logger !== undefined,
-      agentDependencies: this.agentDependencies != undefined,
+      walletConfig: {
+        ...this.walletConfig,
+        key: undefined,
+        storage: { ...this.walletConfig?.storage, credentials: undefined },
+      },
+      logger: this.logger.logLevel,
+      agentDependencies: Boolean(this.agentDependencies),
```
Comment (blu3beri) [inline packages/core/src/agent/AgentConfig.ts:200]:
```suggestion
      walletConfig: {
        ...this.walletConfig,
        key: key ? "[omitted...]" : undefined,
        storage: { ...this.walletConfig?.storage, credentials: this.walletConfig?.credentials ? "[omitted...]" : undefined },
      },
      logger: this.logger.logLevel,
      agentDependencies: Boolean(this.agentDependencies),
```

**Comment by blu3beri** [comment]
@rapaktech Prettier is failing now :). Could you fix that? I can merge it afterwards.

**Comment by blu3beri** [comment]
@TimoGlastra would this be considered a breaking change? The return type of the function is quite different now, but I don't think it is a "public method"

**Review by TimoGlastra — COMMENTED** [review]
I wouldn't call this a breaking change, if you need properties from the agent config you should do so using the class methods. 

This LGTM @rapaktech. I just found another place where we log keys, could you also tackle that one in this PR? It's in the `WalletApi.initialize()` method (`Initializing wallet xxx`)

BTW -- marking this as a fix over chore, as we shouldn't have logged the sensitive info in the first place

**Comment by rapaktech** [comment]
You got it, Timo. On it.

File: packages/core/src/agent/AgentConfig.ts [hunk 3]
```diff
@@ -191,8 +191,16 @@ export class AgentConfig {
...
     return {
       ...this.initConfig,
-      logger: this.logger !== undefined,
-      agentDependencies: this.agentDependencies != undefined,
+      walletConfig: {
+        ...this.walletConfig,
+        key: this.walletConfig?.key ? '[omitted...]' : undefined,
+        storage: {
+          ...this.walletConfig?.storage,
+          credentials: this.walletConfig?.storage?.credentials ? '[omitted...]' : undefined,
```
Comment (jakubkoci) [inline packages/core/src/agent/AgentConfig.ts:199]: This is great and long awaited addition 👍 

Just a naming idea... wouldn't it be better to use rather `[hidden...]` or, quite common use of `*****` (I saw it in Github Action I think)

Comment (rapaktech) [reply packages/core/src/agent/AgentConfig.ts:199]: 👍🏾

**Pull request merged**